src/bake/DevServer.zig

```diff
@@ -6506,6 +6805,371 @@ pub fn onPluginsRejected(dev: *DevServer) !void {
     // TODO: allow recovery from this state
 }
 
+/// Fetched when a client-side error happens. This performs two actions
+/// - Logs the remapped stack trace to the console.
+/// - Replies with the remapped stack trace.
+/// Payload:
+/// - `u32`: Responding message ID (echoed back)
+/// - `u32`: Length of message
+/// - `[n]u8`: Message
+/// - `u32`: Length of error name
+/// - `[n]u8`: Error name
+/// - `u32`: Number of stack frames. For each
+///   - `u32`: Line number (0 for unavailable)
+///   - `u32`: Column number (0 for unavailable)
+///   - `u32`: Length of file name (0 for unavailable)
+///   - `[n]u8`: File name
+///   - `u32`: Length of function name (0 for unavailable)
+///   - `[n]u8`: Function name
+const ErrorReportRequest = struct {
+    dev: *DevServer,
+    body: uws.BodyReaderMixin(@This(), "body", runWithBody, finalize),
+
+    fn run(dev: *DevServer, req: *Request, resp: anytype) void {
+        _ = req;
+
+        const ctx = bun.new(ErrorReportRequest, .{
+            .dev = dev,
+            .body = .init(dev.allocator),
+        });
+        ctx.dev.server.?.onPendingRequest();
+        ctx.body.readBody(resp);
+    }
+
+    fn finalize(ctx: *ErrorReportRequest) void {
+        ctx.dev.server.?.onStaticRequestComplete();
+        bun.destroy(ctx);
+    }
+
+    fn runWithBody(ctx: *ErrorReportRequest, body: []const u8, r: AnyResponse) !void {
+        var s = std.io.fixedBufferStream(body);
+        const reader = s.reader();
+
+        var sfa = std.heap.stackFallback(131072, ctx.dev.allocator);
+        const temp_alloc = sfa.get();
+        var arena = std.heap.ArenaAllocator.init(temp_alloc);
```

Comment:
Memory leak: this is never freed.
```suggestion
        var arena = std.heap.ArenaAllocator.init(temp_alloc);
        defer arena.deinit();
```